builder/ecs/image_config_test.go

```diff
@@ -60,3 +60,83 @@ func TestECSImageConfigPrepare_imageTags(t *testing.T) {
 		}, c.AlicloudImageTags)
 	}
 }
+
+func TestECSImageConfigPrepare_targetImageFamily(t *testing.T) {
+	c := testAlicloudImageConfig()
+
+	// 1 character
+	c.AlicloudTargetImageFamily = "a"
+	if err := c.Prepare(nil); err == nil {
+		t.Fatal("should have error")
+	}
+
+	// 129 character
+	c.AlicloudTargetImageFamily = "abcdefghijklmnopqrs1abcdefghijklmnopqrs2abcdefghijklmnopqrs3abcdefghijklmnopqrs4abcdefghijklmnopqrs5abcdefghijklmnopqrs6123456789"
+	if err := c.Prepare(nil); err == nil {
+		t.Fatal("should have error")
+	}
+
+	// invalid character
+	c.AlicloudTargetImageFamily = "abc%&"
+	if err := c.Prepare(nil); err == nil {
+		t.Fatal("should have error")
+	}
+
+	// start with acs:
+	c.AlicloudTargetImageFamily = "acs:"
+	if err := c.Prepare(nil); err == nil {
+		t.Fatal("should have error")
+	}
+
+	// start with aliyun
+	c.AlicloudTargetImageFamily = "aliyun"
+	if err := c.Prepare(nil); err == nil {
+		t.Fatal("should have error")
+	}
+
+	// start with http://
+	c.AlicloudTargetImageFamily = "http://"
+	if err := c.Prepare(nil); err == nil {
+		t.Fatal("should have error")
+	}
+
+	// start with https://
+	c.AlicloudTargetImageFamily = "https://"
+	if err := c.Prepare(nil); err == nil {
+		t.Fatal("should have error")
+	}
+
+	// success
+	c.AlicloudTargetImageFamily = "success"
```

Comment:
Just to be sure the `\p{L}` expression captures that, mind adding a test with chinese characters since they're explicitly documented as valid for the URIs?

Reply:
I add new tests to ensure the regex can match Chinese characters, -, _, : and .